lib/src/room.dart

```diff
@@ -211,37 +218,48 @@ class KurentoRoom extends ChangeNotifier implements Room {
     if (_localStream == null) {
       throw StateError('Cannot mute audio before joining the room');
     }
-
-    _localStream!.getAudioTracks()[0].enabled = !muted;
-    _updateParticipantStatus();
+    if (_localStream!.getAudioTracks().isEmpty) {
+      _log.warning('Cannot mute audio because there is not audio tracks to mute');
+    } else {
+      _localStream!.getAudioTracks()[0].enabled = !muted;
+      _updateParticipantStatus();
+    }
   }
 
   @override
   void setVideoMuted(bool muted) {
     if (_localStream == null) {
       throw StateError('Cannot mute video before joining the room');
     }
-
-    _localStream!.getVideoTracks()[0].enabled = !muted;
-    _updateParticipantStatus();
+    MediaStreamTrack? activeVideoTrack = _activeVideoTrack;
+    if (null == activeVideoTrack) {
+      _log.warning('Cannot mute video because there is not video tracks to mute');
+    } else {
+      activeVideoTrack.enabled = !muted;
+      _updateParticipantStatus();
+    }
   }
 
   @override
   Future<void> startPresenting(MediaStream displayStream) async {
     // Eventually, we will create a separate connection for screen sharing. That requires changes to Platform and Dash,
     // so for now, we start presenting by replacing the Explorer's video track with the display video track.
-    await _connectionByTrackId[_localTrackId]!.replaceTrack(displayStream.getVideoTracks()[0]);
-    _presenting = true;
-    _log.info('started presenting track=${displayStream.getVideoTracks()[0].label}');
+    _presentationVideoTrack = displayStream.getVideoTracks()[0];
+    await _connectionByTrackId[_localTrackId]!.replaceVideoTrack(_presentationVideoTrack);
+    _log.info('started presenting track=${_presentationVideoTrack!.label}');
+    await _updateParticipantStatus();
```

Comment:
I think adding mute support for screen sharing is adding unnecessary complexity. Try this in the Aira Desktop app:
1. Start screen sharing.
2. Disable camera. This will mute the screen share and will toggle the button to enable camera. It will also say "camera disabled" even though it's not.
3. Stop sharing screen. Dash will report that the camera is enabled, but it doesn't appear to be.

Reply:
I had in mind to change that message when we are presenting, but forgot. I'll keep my notepad handy to avoid this in the future.